### PR TITLE
Banking system tests

### PR DESCRIPTION
**Pull request by BEEG3R**
Review of current implementation of Banking related objects, with some aspects incomplete. Since we do not have any Cards or Properties implemented, anything related to them in the current implementation uses (void *). There should be more test cases added to the existing test projects here once Cards and Properties are implemented.

### CODE REVIEW

File: Tests/TestTransactions/TestTransactions.pro [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+CONFIG += qt console warn_on depend_includepath testcase
+CONFIG -= app_bundle
+
+TEMPLATE = app
+
+INCLUDEPATH += ../../src/Model
+
+SOURCES +=  tst_testtransaction.cpp \
+    ../../src/Model/Banking/Transaction.cpp \
```
Comment (kornawe) [inline Tests/TestTransactions/TestTransactions.pro:12]: Yeah, I don't like it, but I think it's the most feasible way to do the tests

File: Tests/TestTransactions/tst_testtransaction.cpp [hunk 2]
```diff
@@ -0,0 +1,160 @@
+#include "tst_testtransaction.h"
+
+TestTransaction::TestTransaction() {
+    CuT = new ITransaction(nullptr, nullptr, nullptr);
+
+    amount = new IAmount(0, nullptr, nullptr);
+    acct_one = new PlayerAccount(0);
```
Comment (kornawe) [inline Tests/TestTransactions/tst_testtransaction.cpp:10]: Should we make a Mock Account??

File: Tests/TestTransactions/tst_testtransaction.cpp [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+
+    delete amount;
+    delete acct_one;
+    delete acct_two;
+}
+
+///
+/// \brief TestTransaction::testAmountGetter
+/// Verify that the Amount accessor works as intended.
+void TestTransaction::testAmountGetter() {
```
Comment (kornawe) [inline Tests/TestTransactions/tst_testtransaction.cpp]: I think we should shoot for the test naming convention at work. function_abcShouldxyz

Comment (BEEG3R) [reply Tests/TestTransactions/tst_testtransaction.cpp]: I agree. I'll make that change.

File: Tests/TestTransactions/tst_testtransaction.cpp (hunk 2 above)
Comment (BEEG3R) [reply Tests/TestTransactions/tst_testtransaction.cpp:10]: Certainly. Hadn't occurred to me when writing, but I'll make some Fakes.

**Pull request closed without merging**